dipy/denoise/localpca.py

```diff
@@ -139,11 +139,16 @@ def genpca(arr, sigma=None, mask=None, patch_radius=2, pca_method='eig',
     else:
         raise ValueError("pca_method should be either 'eig' or 'svd'")
 
+    if isinstance(patch_radius, int):
+        patch_radius = np.ones(3) * patch_radius
+    if len(patch_radius) != 3:
+        raise ValueError("patch_radius should have length 3")
     patch_size = 2 * patch_radius + 1
 
-    if patch_size ** 3 < arr.shape[-1]:
+    if np.prod(patch_size) < arr.shape[-1]:
         e_s = "You asked for PCA denoising with a "
         e_s += "patch_radius of {0} ".format(patch_radius)
+        e_s += "with total patch size of {0}".format(np.prod(patch_size))
         e_s += "for data with {0} directions. ".format(arr.shape[-1])
         e_s += "This would result in an ill-conditioned PCA matrix. "
         e_s += "Please increase the patch_radius."
```

Comment:
Is this the kind of unit test you were thinking of? @ShreyasFadnavis

Reply:
Yep! Thanks!

Reply:
You will also need to adapt the tests like:
https://github.com/nipy/dipy/blob/9f3dc9b58e3b6b7390c08968c07822a28c0fe1f3/dipy/denoise/tests/test_lpca.py#L242

With the new type of patch radius that you have proposed... does this make sense?

Reply:
> You will also need to adapt the tests like:
> https://github.com/nipy/dipy/blob/9f3dc9b58e3b6b7390c08968c07822a28c0fe1f3/dipy/denoise/tests/test_lpca.py#L242
> 
> With the new type of patch radius that you have proposed... does this make sense?

See the whole file.. that will make Travis happy👌🏻